src/FileIO.cpp

```diff
@@ -113,7 +113,13 @@ namespace FileIO {
       // Attempt to read it the fastest way possible.
       if (-1 != end) {  // tellg() --> pos_type{-1} if reading the end failed.
          contents.resize(end);
-         in.read(&contents[0], contents.size());
```

Comment:
not reviewing this,. just interested in understanding it. 
(The downfall of having public repos is that people can read the PRs ;)

Is the exception triggered if the file is moved or deleted during a read? Linux has a `read count` on the file system so if you delete a file, as the file is being used, the process using it will still have access to it. It's only `deleted` once the read count goes to zero.

Reply:
That is a good question, I am not sure if that triggers the exception or not. This `ifstream::read()` threw an (uncaught) exception one time inexplicably, and we haven't been able to reproduce it since. Better safe than sorry though 😸

Reply:
Agreed. Not sure how that can be unit tested either